pkg/proxy/winkernel/proxier.go

```diff
@@ -204,6 +213,14 @@ func newServiceInfo(svcPortName proxy.ServicePortName, port *v1.ServicePort, ser
 	if service.Spec.SessionAffinity == v1.ServiceAffinityClientIP && service.Spec.SessionAffinityConfig != nil {
 		stickyMaxAgeSeconds = int(*service.Spec.SessionAffinityConfig.ClientIP.TimeoutSeconds)
 	}
+
+	klog.Infof("Service %q preserve-destination: %v", svcPortName.NamespacedName.String(), service.Annotations["preserve-destination"])
+
+	preserveDIP := service.Annotations["preserve-destination"] == "true"
```

Comment:
cc @kubernetes/sig-network-api-reviews 

is this annotation-driven behavior novel for in-tree components?

Reply:
Currently this is specific to windows, but can be extendend to general kube-proxy and can be implemented across.